### DISCUSSION TITLE
Custom status messages/descriptions with commas/slashes

### DISCUSSION THREAD

**Discussion by avm99963** [tarampampam/error-pages#297] [General] [answered]
Hi!

First of all, I'd like to thank you @tarampampam for building this! I'm using error-pages for my homelab Traefik instance and I like it very much :)

For a long time I've been customizing all the status messages/descriptions via the config file, and some include commas and slashes.

When the v3 update came out, I began trying to convert these definitions to the `--add-code` flag equivalents, but I noticed that using commas and slashes caused errors when parsing the flags because:

1. By default [urfave/cli](https://pkg.go.dev/github.com/urfave/cli/v3) treats commas as a separator for the different elements of the string map.
2. Slashes are used for separating the message from the description.

I wanted to share that I worked around this by replacing these special characters in the flag with `##comma##` and `##slash##`. Then, in my custom template I replace these special strings back to actual commas and slashes using the `strReplace` function which you made available:

```
{{ strReplace (strReplace message "##slash##" "/") "##comma##" "," }}
```

It's not ideal, but I can continue using commas and slashes, so I'm happy with that ;)

**Comment by tarampampam** [comment, marked as answer]
https://github.com/tarampampam/error-pages/blob/11aebaa4d6c577c810db9e20f620aeaf690f2116/docs/UPGRADE_TO_V4.md?plain=1#L108-L128
